contracts/oracles/curve/CurveTWAPPriceFeed.sol

```diff
@@ -0,0 +1,109 @@
+contract CurveTWAPPriceFeed is IPriceFeed, PriceFeedValidationTrait, SanityCheckTrait {
+    using Address for address;
+
+    /// @notice Thrown when the curve oracle value is out of bounds
+    error CurveOracleOutOfBoundsException();
+
+    /// @notice Thrown when the upper bound is less than the lower bound passed in constructor
+    error UpperBoundTooLowException();
+
+    /// @notice Contract version
+    uint256 public constant override version = 3_10;
+
+    /// @notice Contract type
+    bytes32 public constant override contractType = "PRICE_FEED::CURVE_TWAP";
+
+    /// @notice Answer precision (always 8 decimals for USD price feeds)
+    uint8 public constant override decimals = 8;
+
+    /// @notice Indicates that price oracle can skip checks for this price feed's answers
+    bool public constant override skipPriceCheck = true;
+
+    /// @notice token token address
+    address public immutable token;
+
+    /// @notice Curve pool address
+    address public immutable pool;
+
+    /// @notice Coin 0 price feed address
+    address public immutable priceFeed;
+
+    /// @notice Staleness period for the coin 0 price feed
+    uint32 public immutable stalenessPeriod;
+
+    /// @notice Flag indicating if price feed checks can be skipped
+    bool public immutable skipCheck;
+
+    /// @notice Lower bound for the exchange rate
+    uint256 public immutable lowerBound;
+
+    /// @notice Upper bound for the exchange rate
+    uint256 public immutable upperBound;
+
+    /// @notice Price feed description
+    string public description;
+
+    /// @notice Constructor
+    /// @param _lowerBound Lower bound for the pool exchange rate
+    /// @param _upperBound Upper bound for the pool exchange rate (must be greater than lower bound)
+    /// @param _token Address of the coin 1 in the pool
+    /// @param _pool Address of the curve pool
+    /// @param _priceFeed Address of the coin 0 price feed
+    /// @param _stalenessPeriod Staleness period for the coin 0 price feed
+    constructor(
+        uint256 _lowerBound,
+        uint256 _upperBound,
+        address _token,
+        address _pool,
+        address _priceFeed,
+        uint32 _stalenessPeriod
+    ) nonZeroAddress(_token) nonZeroAddress(_pool) nonZeroAddress(_priceFeed) {
+        if (_upperBound < _lowerBound) revert UpperBoundTooLowException();
+
+        token = _token;
+        pool = _pool;
+        priceFeed = _priceFeed;
+        stalenessPeriod = _stalenessPeriod;
+        skipCheck = _validatePriceFeedMetadata(_priceFeed, _stalenessPeriod);
+        lowerBound = _lowerBound;
+        upperBound = _upperBound;
+
+        string memory tokenSymbol = IERC20Metadata(_token).symbol();
+        description = string.concat(tokenSymbol, " / USD Curve TWAP price feed");
+    }
+
+    /// @notice Serialized price feed parameters
+    function serialize() external view override returns (bytes memory) {
+        return abi.encode(token, pool, priceFeed, lowerBound, upperBound);
```

Comment:
We can remove `priceFeed` from here I guess:
- serialization usually includes only "non-standard" fields (and stuff like `priceFeed` and `stalenessPeirod` is handled by `NestedPriceFeeds` lib in compressors)
- if we do, we get the matching signature to `LPPriceFeed`'s serialization:

https://github.com/Gearbox-protocol/oracles-v3/blob/fc8d3a0ab5bd7eb50ce3f6b87dde5cd3d887bafe/contracts/oracles/LPPriceFeed.sol#L66-L70